backend/plugin/db/mongodb/mongodb.go

```diff
@@ -397,169 +393,25 @@ func getSimpleStatementResult(data []byte) (*v1pb.QueryResult, error) {
-			case primitive.ObjectID:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: value.String()}}
-			case primitive.Regex:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: value.String()}}
-			case primitive.JavaScript:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: string(value)}}
-			case primitive.CodeWithScope:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: value.String()}}
-			case primitive.Undefined, primitive.Null:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_NullValue{}}
-			case primitive.Symbol:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: string(value)}}
-			case primitive.DBPointer:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: value.String()}}
-			case int32:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_Int32Value{Int32Value: value}}
-			case int64:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_Int64Value{Int64Value: value}}
-			case string:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: value}}
-			case float64:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_DoubleValue{DoubleValue: value}}
-			case bool:
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_BoolValue{BoolValue: value}}
-			case primitive.D, primitive.M:
-				r, err := bson.MarshalExtJSONIndent(value, true, false, "", "	")
-				if err != nil {
-					return nil, errors.Wrapf(err, "failed to marshal ejson for document")
-				}
-				index := columnIndexMap[k]
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: string(r)}}
-
-			case primitive.A:
-				// bson.MarshalExtJSONIndent doesn't allow marshal array directly.
-				// we compose the array into a document, marshal the document,
-				// then extract the array using jsonparser library.
-				ejson, err := bson.MarshalExtJSONIndent(primitive.D{{Key: "array", Value: value}}, true, false, "", "	")
-				if err != nil {
-					return nil, errors.Wrapf(err, "failed to marshal ejson for array")
-				}
-
-				s, _, _, err := jsonparser.Get(ejson, "array")
-				if err != nil {
-					return nil, errors.Wrapf(err, "failed to get string")
-				}
-
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: string(s)}}
-
-			// primitive.Timestamp
-			// primitive.MinKey, primitive.MaxKey
-			default:
-				r, err := json.MarshalIndent(v, "", "	")
-				if err != nil {
-					return nil, err
-				}
-				index := columnIndexMap[k]
-				values[index] = &v1pb.RowValue{Kind: &v1pb.RowValue_StringValue{StringValue: string(r)}}
-			}
-		}
-		for i := 0; i < len(values); i++ {
-			if values[i] == nil {
-				values[i] = &v1pb.RowValue{Kind: &v1pb.RowValue_NullValue{}}
-			}
+		r, err := json.MarshalIndent(v, "", "	")
```

Comment:
https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#MarshalExtJSON

Use `MarshalExtJSON` with `canonical=true` to keep type info.

Reply:
A regression test case: try testing with an int64 value bigger than 2^53.

Reply:
LGTM